src/main/java/org/springframework/data/elasticsearch/core/mapping/IndexInformation.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.data.elasticsearch.core.mapping;
+
+import java.util.List;
+
+import org.elasticsearch.cluster.metadata.AliasMetadata;
+import org.springframework.data.elasticsearch.core.document.Document;
+import org.springframework.lang.Nullable;
+
+/**
+ * Immutable object that holds information(name, settings, mappings, aliases) about an Index
+ *
+ * @author George Popides
+ * @since 4.2
+ */
+public class IndexInformation {
+	private final String name;
+	@Nullable
+	private final Document settings;
+	@Nullable
+	private final Document mappings;
+	@Nullable
+	private final List<AliasMetadata> aliases;
```

Comment:
I just now noticed that here we use `AliasMetadata` from Elasticsearch. We should use `org.springframework.data.elasticsearch.core.index.AliasData`, the same class that is used in the return values of `org.springframework.data.elasticsearch.core.AbstractDefaultIndexOperations.getAliasesForIndex()`

Reply:
I'm searching for the way to conver ```AliasMetadata``` to ```AliasData``` but something looks off.

The ```filter``` atribute of ```AliasData``` is ```Document``` which essentialy is a ```Map```, but the ```filter``` attribute of ```AliasMetadata``` is ```CompressedXContent```.  So how do i convert it to a Document?

Reply:
there already is `org.springframework.data.elasticsearch.core.RequestFactory.convertAliasMetadata()` - which probably could be moved to `ResponseConverter` as well